$OpenBSD: patch-src_racket_src_fun_c,v 1.1 2016/03/10 02:35:29 juanfra Exp $

"avoid some C undefined behavior"
https://github.com/racket/racket/commit/9a8fd29

"Avoid compiler warnings"
https://github.com/racket/racket/commit/65eaff3

"fix mishandling of the continuation-mark depth"
https://github.com/racket/racket/commit/81b5d74ed695b4d73b341194ccb77128b54a3fef

--- src/racket/src/fun.c.orig	Sun Jan 24 16:39:49 2016
+++ src/racket/src/fun.c	Sun Feb 28 00:33:35 2016
@@ -1202,7 +1202,7 @@ void scheme_set_dynamic_state(Scheme_Dynamic_State *st
   state->menv              = menv;
 }
 
-static void *apply_again_k()
+static void *apply_again_k(void)
 {
   Scheme_Thread *p = scheme_current_thread;
   Scheme_Object *val = p->ku.k.p1;
@@ -2587,7 +2587,7 @@ Scheme_Object *scheme_get_or_check_procedure_shape(Sch
     /* Integer encoding, but shift to use low bit to indicate whether
        it preserves marks, which is useful information for the JIT. */
     intptr_t i = SCHEME_INT_VAL(p);
-    i <<= 1;
+    i = ((uintptr_t)i) << 1;
     if (scheme_closure_preserves_marks(e)) {
       i |= 0x1;
     }
@@ -3640,7 +3640,7 @@ static Scheme_Object *_apply_native(Scheme_Object *obj
   obj = data->start_code(obj, num_rands, rands EXTRA_NATIVE_ARGUMENT);
 
   if (obj == SCHEME_TAIL_CALL_WAITING)
-    obj = force_values(obj, 1);
+    obj = scheme_force_value_same_mark(obj);
 
   MZ_CONT_MARK_STACK = old_cont_mark_stack;
   MZ_CONT_MARK_POS -= 2;
@@ -9834,7 +9834,8 @@ static Scheme_Object *seconds_to_date(int argc, Scheme
 {
   UNBUNDLE_TIME_TYPE lnow;
   int get_gmt;
-  int hour, min, sec, month, day, year, wday, yday, dst;
+  int hour, min, sec, month, day, wday, yday, dst;
+  intptr_t year;
   long tzoffset;
 #ifdef USE_WIN32_TIME
 # define CHECK_TIME_T uintptr_t
@@ -9959,7 +9960,7 @@ static Scheme_Object *seconds_to_date(int argc, Scheme
 
       month = localTime->tm_mon + 1;
       day = localTime->tm_mday;
-      year = localTime->tm_year + 1900;
+      year = (uintptr_t)localTime->tm_year + 1900;
 
       wday = localTime->tm_wday;
       yday = localTime->tm_yday;
